packages/calcite-components/src/components/shell-center-row/shell-center-row.tsx

```diff
@@ -68,28 +51,33 @@ export class ShellCenterRow implements ConditionalSlotComponent {
   // --------------------------------------------------------------------------
 
   render(): VNode {
-    const { el } = this;
+    const { actionBar } = this;
 
     const contentNode = (
       <div class={CSS.content}>
         <slot />
       </div>
     );
 
-    const actionBar = getSlotted<HTMLCalciteActionBarElement>(el, SLOTS.actionBar);
-
-    const actionBarNode = actionBar ? (
-      <div class={CSS.actionBarContainer} key="action-bar">
-        <slot name={SLOTS.actionBar} />
+    const actionBarNode = (
+      <div class={CSS.actionBarContainer} hidden={!this.actionBar} key="action-bar">
+        <slot name={SLOTS.actionBar} onSlotchange={this.handleActionBarSlotChange} />
       </div>
-    ) : null;
+    );
 
     const children: VNode[] = [actionBarNode, contentNode];
 
+    // todo: if actionBar position changes, this will not update.
     if (actionBar?.position === "end") {
```

Comment:
if actionBar position changes, this will not update. Since this component is deprecated I guess it doesn't matter too much.

Reply:
https://github.com/Esri/calcite-design-system/issues/10468